context/src/env/parser.rs

```diff
@@ -392,6 +443,54 @@ impl CIInfo {
     }
 }
 
+// unfortunately magnus requires explicit getters and setters for the object
+// to be accessible from Ruby
```

Comment:
Probably doesn't need to be a code comment since I'm assuming we'll be doing this all over the place. I agree that this is a lot. There must be a nice proc macro someone has written to help do this the same way that `pyo3` or `wasm-bindgen` have done. At worse, we can try to implement our own in the future.